module/applications/item/item-sheet.mjs

```diff
@@ -610,6 +619,53 @@ export default class ItemSheet5e extends ItemSheet {
 
   /* -------------------------------------------- */
 
+  /**
+   * Handle the dropping of an advancement or item with advancements onto the advancements tab.
+   * @param {DragEvent} event                  The concluding DragEvent which contains drop data.
+   * @param {object} data                      The data transfer extracted from the event.
+   */
+  async _onDropAdvancement(event, data) {
+    let advancements;
+    let showDialog = false;
+    if ( data.type === "Advancement" ) {
+      advancements = [await fromUuid(data.uuid)];
+    } else if ( data.type === "Item" ) {
+      const item = await Item.implementation.fromDropData(data);
+      if ( !item ) return false;
+      advancements = Object.values(item.advancement.byId);
+      showDialog = true;
+    } else {
+      return false;
+    }
+    advancements = advancements.filter(a => {
+      return !this.item.advancement.byId[a.id]
+        && a.constructor.metadata.validItemTypes.has(this.item.type)
+        && a.constructor.availableForItem(this.item);
+    });
+
+    // Display dialog prompting for which advancements to add
+    if ( showDialog ) {
+      try {
+        advancements = await AdvancementMigrationDialog.createDialog(this.item, advancements);
+      } catch(err) {
+        return false;
+      }
+    }
+
+    if ( !advancements.length ) return false;
+    if ( this.item.isEmbedded && !game.settings.get("dnd5e", "disableAdvancements") ) {
```

Comment:
Shouldn't we have this check for disabled advancements much earlier? If advancements are disabled we can bail out right at the top of the method, no?

Reply:
We still want to copy over the advancements to the item even if advancements are disabled, we just don't want to launch the `AdvancementManager` to apply their changes to the actor.

Reply:
I suppose that does make sense; fair enough then